File: src/medium/IsSymmetric.java
package medium;

import common.TreeNode;

// https://leetcode.com/problems/symmetric-tree/
public class IsSymmetric {
  public boolean isSymmetric(TreeNode root) {
    return root == null || helper(root.left, root.right);
  }

  private boolean helper(TreeNode left, TreeNode right) {
    if (left == null || right == null) {
      return left == right;
    }
    return left.val == right.val && helper(left.right, right.left) && helper(left.left, right.right);
  }
}

File: csharp/LeetCode/Easy/IsSymmetric.cs
namespace LeetCode.Easy
{
    using LeetCode.DataStructures;

    public partial class Solution
    {
        public bool IsSymmetric(TreeNode root)
        {
            if (root == null)
            {
                return true;
            }

            return IsSymmetric(root.left, root.right);
        }

        bool IsSymmetric(TreeNode left, TreeNode right)
        {
            if (left == null && right == null)
            {
                return true;
            }
            if (left == null || right == null)
            {
                return false;
            }
            if (left.val != right.val)
            {
                return false;
            }
            return IsSymmetric(left.left, right.right) && IsSymmetric(left.right, right.left);
        }
    }
}
